main.cpp

```diff
@@ -6,25 +6,62 @@
 
 #include "xos.h"
 
-int main(int argc, char **argv) {
-  if (argc < 2) {
+int main(int argc, char **argv)
+{
+  if (argc < 2)
+  {
     std::cerr << "Missing input file" << std::endl;
     return 1;
   }
 
   std::ifstream input(argv[1]);
   xos::Lexer lexer(input);
-  while (1) {
-    xos::Result<xos::Token> tok = lexer.getNextToken();
-    if (tok.hasError()) {
+  xos::Result<xos::Token> tok{xos::Token::eof, 0, 0, xos::Token::val{.eof = true}}; // set to a random value -- for now for testing.
+  while (1)
+  {
+    tok = lexer.getNextToken();
+    if (tok.hasError())
+    {
       std::cerr << tok.getErr() << std::endl;
       return 1;
     }
-    std::cout << tok.get().getKind() << " , ";
+    auto tok_kind = tok.get().getKind();
+    if (tok.get().getKind() == xos::Token::eof)
+    {
+      std::cout << tok.get().getVal().eof << std::endl;
+      break;
+    }
 
-    if (tok.get().getKind() == xos::Token::eof) break;
+    switch (tok_kind)
```

Comment:
Rather than having this switch, you can do something like `std::cout << tok.get().getVal() << std::endl` once the comment below is addressed.